public/app/plugins/datasource/grafana-pyroscope-datasource/dataquery.cue

```diff
@@ -31,6 +31,8 @@ composableKinds: DataQuery: {
 
 				// Specifies the query label selectors.
 				labelSelector: string | *"{}"
+				// Specifies the query span selectors.
+				spanSelector?: [...string]
```

Comment:
Similar thing for the typing, if the span and labels cannot match this should be two separate query types.